File: 03-tobogganTrajectory/go/challenge/partTwo.go
package challenge

func PartTwo(instr string) int {
	forest := parse(instr)

	offsetPairs := [][]int{
		{3, 1},
		{1, 1},
		{5, 1},
		{7, 1},
		{1, 2},
	}

	treeProduct := 1

	for _, pair := range offsetPairs {
		encounteredTrees := findCollisions(forest, pair[0], pair[1])
		treeProduct *= encounteredTrees
	}

	return treeProduct
}

File: challenges/2020/03-tobogganTrajectory/python/partTwo.py
from common import *


def partTwo(instr: str) -> int:
    forest = parse(instr)

    tree_product = 1

    offset_pairs = [(3, 1), (1, 1), (5, 1), (7, 1), (1, 2)]

    for i, pair in enumerate(offset_pairs):
        encountered_trees = find_collisions(forest, *pair)
        tree_product *= encountered_trees

    return tree_product
